Modules/timemodule.c

```diff
@@ -1222,8 +1300,17 @@ PyInit_timezone(PyObject *m) {
     PyModule_AddIntConstant(m, "altzone", timezone-3600);
 #endif
     PyModule_AddIntConstant(m, "daylight", daylight);
+    /*bpo-16322, bpo-27426: For Windows use GetTimeZoneInformation() to avoid
+    wrong encoding of time zone names.*/
+#ifdef MS_WINDOWS
+    TIME_ZONE_INFORMATION tzi;
+    GetTimeZoneInformation(&tzi);
```

Comment:
Handle error (TIME_ZONE_ID_INVALID).

Is `DaylightName` initialized properly if `GetTimeZoneInformation() != TIME_ZONE_ID_DAYLIGHT`?

Reply:
Yes, I think so. TIME_ZONE_INFORMATION structure get all members if GetTimeZoneInformation() succeeds. On my system GetTimeZoneInformation() == TIME_ZONE_ID_UNKNOWN, i.e. daylight saving time is not used, but both names initialized properly. However, DaylightName and StandardName can be empty.